cdap-gateway/src/main/java/io/cdap/cdap/gateway/router/handlers/AuthenticationHandler.java

```diff
@@ -111,7 +117,10 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
       UserIdentityPair userIdentityPair = extractionResponse.getIdentityPair();
       // User identity extraction succeeded, so set some header properties and allow the call through
       request.headers().remove(HttpHeaderNames.AUTHORIZATION);
-      String credential = userIdentityPair.getUserCredential();
+
+      // When enabled, encrypt user credential and pass it in the encrypted form around services.
+      String credential = encryptUserCredential(userIdentityPair.getUserCredential());
```

Comment:
may be rename this to `encryptedCredential` for better code readability

Reply:
Depending on if encryption is enabled or not, this may or may not be encrypted credential.

Reply:
Maybe call the method as `getUserCredential(userIdentityPair)`

Reply:
Code updated. PTAL